onnxruntime/core/session/IOBinding.cc

```diff
@@ -11,32 +11,23 @@ namespace onnxruntime {
 IOBinding::IOBinding(const SessionState& session_state) : session_state_(session_state) {
 }
 
-static std::pair<bool, size_t> Contains(const std::vector<std::string>& names, const std::string& name) {
-  auto it = std::find(std::begin(names), std::end(names), name);
-  if (it == std::end(names)) {
-    return {false, 0};
-  }
-  return {true, it - std::begin(names)};
-}
-
 common::Status IOBinding::BindInput(const std::string& name, const OrtValue& ml_value) {
-  auto rc = Contains(feed_names_, name);
-
-  auto add_or_replace = [this, &name](const bool exists, size_t index, const OrtValue& value) {
-    if (exists) {
-      feeds_[index] = value;
-    } else {
-      feed_names_.push_back(name);
-      feeds_.push_back(value);
-    }
-  };
+  auto it = mapped_feed_names_.find(name);
+  size_t index;
+  if (it == mapped_feed_names_.end()) {
```

Comment:
Can you do insert once and check if the value has been inserted to avoid double lookup?